### PR TITLE
Refractor `giveaway.py` into a package based cog

### PR DESCRIPTION
**Pull request by ShivamGarg2910**
## Checklist
- [x] I have read the [contributing guidelines](./CONTRIBUTING.md).
- [x] The changes are for an issue, or have been discussed with Bluesy.
- [x] I have checked the code for syntax errors or non conforming style and fixed them to the best of my ability.

###  Changes

refactoring giveaway.py into a package.

Closes: 237

### Pending items
- Python version not mentioned. @Bluesy1: pleas share python version, getting syntax errors.
- Testcases pending, waiting for python version

### CODE REVIEW

**Comment by Bluesy1** [comment]
@ShivamGarg2910 I target python3.10

**Review by Bluesy1 — CHANGES_REQUESTED** [review]
the setup function should be in `__init__.py`, not `cog.py`, like here: https://github.com/Bluesy1/CharB0T/blob/890c5d1515987bd8e051ff538b8cf140ea9e81a9/charbot/programs/__init__.py#L16-L23

Besides that and test coverage, LGTM!

File: charbot/giveaway/view.py
Comment (Bluesy1) [inline charbot/giveaway/view.py:1]: Need to run black on this file.

File: charbot/giveaway/cog.py
Comment (Bluesy1) [inline charbot/giveaway/cog.py:1]: Need to run black on this file

**Review by Bluesy1 — CHANGES_REQUESTED** [review]
See comments + please rebase off of main.

**Comment by Bluesy1** [comment]
@ShivamGarg2910 i'd like to push this through in the next week or so, preferably before hacktoberfest ends too.

Any update?

**Comment by ShivamGarg2910** [comment]
Hey @Bluesy1, have been a little under the weather, will try finishing it up asap.

**Comment by Bluesy1** [comment]
@ShivamGarg2910 ah no worries, feel better soon

**Comment by ShivamGarg2910** [comment]
Hey @Bluesy1, please help by confirming that pending tasks are running the black commands and moving setup function to __init__.py

File: charbot/giveaway/view.py
Comment (ShivamGarg2910) [reply charbot/giveaway/view.py:1]: resolved

**Review by Bluesy1 — DISMISSED** [review]
LGTM! @ShivamGarg2910 please just resolve the merge conflict (or let me know if you want me to, i'm happy to do it because its a complex one)

**Comment by ShivamGarg2910** [comment]
@Bluesy1, can you please resolve the merge conflict, and share some guide on how to do the same.
Coz I tried the rebase and merge, but no conflicts came.

**Comment by Bluesy1** [comment]
@ShivamGarg2910 i've taken care of the merge/rebase
Unfortunately i don't have a great guide to show you besides [github's docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github), but what i did here was i clone your fork into my IDE (integrated development environment), in py case PyCharm, which is free for students via teh Github Education Pack, and told it to checkout and rebase your branch onto main (in hindsight, a merge should have been done over a rebase, apologies for that). 
Git clients can help with this, like Github Desktop, GitKraken, even the one in VSCode
Once CI passes, this is now ready to merge, thanks for your contributon!

**Review by Bluesy1 — APPROVED** [review]
LGTM! Waiting for CI to pass

File: charbot/giveaway/cog.py
Comment (Bluesy1) [inline charbot/giveaway/cog.py:17]:
```suggestion
if TYPE_CHECKING:
    from . import GiveawayView
```

File: charbot/giveaway/cog.py
Comment (Bluesy1) [inline charbot/giveaway/cog.py:7]:
```suggestion
"""Giveaway  commands."""
from __future__ import annotations
```

File: charbot/giveaway/cog.py
Comment (Bluesy1) [inline charbot/giveaway/cog.py:9]:
```suggestion
from typing import TYPE_CHECKING, cast
```

File: charbot/giveaway/cog.py
Comment (Bluesy1) [inline charbot/giveaway/cog.py:6]:
```suggestion
```

**Comment by Bluesy1** [comment]
Thanks for your contribution!

**Pull request merged**
